liminal/runners/airflow/dag/liminal_dags.py

```diff
@@ -15,105 +15,135 @@
+    logging.info(f'found {len(configs)} liminal configs in path: {configs_path}')
+    for config in configs:
+        name = config['name'] if 'name' in config else None
+        try:
+            if not name:
+                raise ValueError('liminal.yml missing field `name`')
 
-                default_args = {k: v for k, v in pipeline.items()}
+            logging.info(f"Registering DAGs for {name}")
 
-                override_args = {
-                    'start_date': datetime.combine(pipeline['start_date'], datetime.min.time()),
-                    __DEPENDS_ON_PAST: default_args[
-                        __DEPENDS_ON_PAST] if __DEPENDS_ON_PAST in default_args else False,
-                }
+            owner = config.get('owner')
 
-                default_args.update(override_args)
+            trigger_rule = 'all_success'
+            if 'always_run' in config and config['always_run']:
+                trigger_rule = 'all_done'
 
-                dag = DAG(
-                    dag_id=pipeline_name,
-                    default_args=default_args,
-                    dagrun_timeout=timedelta(minutes=pipeline['timeout_minutes']),
-                    catchup=False
-                )
-
-                job_start_task = JobStartTask(dag, config, pipeline, {}, None, 'all_success')
-                parent = job_start_task.apply_task_to_dag()
+            for pipeline in config['pipelines']:
+                default_args = __default_args(pipeline)
+                dag = __initialize_dag(default_args, pipeline, owner)
 
-                trigger_rule = 'all_success'
-                if 'always_run' in config and config['always_run']:
-                    trigger_rule = 'all_done'
+                parent = None
 
                 for task in pipeline['tasks']:
                     task_type = task['type']
                     task_instance = get_task_class(task_type)(
-                        dag, config, pipeline, task, parent if parent else None, trigger_rule
+                        task_id=task['task'],
+                        dag=dag,
+                        parent=parent,
+                        trigger_rule=trigger_rule,
+                        liminal_config=config,
+                        pipeline_config=pipeline,
+                        task_config=task
                     )
 
                     parent = task_instance.apply_task_to_dag()
 
-                job_end_task = JobEndTask(dag, config, pipeline, {}, parent, 'all_done')
-                job_end_task.apply_task_to_dag()
-
                 logging.info(f'registered DAG {dag.dag_id}: {dag.tasks}')
 
-                globals()[pipeline_name] = dag
+                globals()[pipeline['pipeline']] = dag
                 dags.append(dag)
 
+        except Exception:
+            logging.error(f'Failed to register DAGs for {name}')
+            traceback.print_exc()
+
     return dags
 
 
+def __initialize_dag(default_args, pipeline, owner):
```

Comment:
Will it be useful if these default behaviors will reside in the super/hyper liminal definitions?

Reply:
not sure what do u mean. 
note that the code here working to the dags and not the configuration of the liminal.